contracts/farms/RevoMobiusFarmBot.sol

```diff
@@ -0,0 +1,217 @@
+
+    IERC20 public stakingToken0;
+    IERC20 public stakingToken1;
+
+    IERC20[] public rewardsTokens;
+
+    uint8 public celoNativeStakingTokenIndex;
+    uint8 public bridgedStakingTokenIndex;
+
+    IERC20 public celoNativeStakingToken;
+
+    ILiquidityGaugeV3 public liquidityGauge; // The Mobius Liquidity Gauge that mints rewards
+    IMinter public minter; // The Minter contract through which Mobius rewards are claimed
+    IUniswapV2Router02SwapOnly public router; // A router capable of swapping reward for the stakingToken's Celo-native constituent asset
+    ISwap public swap; // The Mobius Swap contract used to swap LP constituent tokens and mint/burn liquidity
+
+    // ISwap's `addLiquidity` method requires a parameter located in calldata, but it must be computed dynamically, and thus cannot
+    // be provided by the compounder. This private variable is used as a dynamic array that we can pass as calldata.
+    uint256[] private amounts;
+
+    constructor(
+        address _owner,
+        address _reserveAddress,
+        address _stakingToken,
+        address _revoFees,
+        address[] memory _rewardsTokens,
+        address _liquidityGauge,
+        address _minter,
+        address _router,
+        address _swap,
+        uint8 _celoNativeStakingTokenIndex,
+        string memory _symbol
+    )
+        StakingTokenHolder(
+            _owner,
+            _reserveAddress,
+            _stakingToken,
+            _revoFees,
+            _symbol
+        )
+    {
+        liquidityGauge = ILiquidityGaugeV3(_liquidityGauge);
+        router = IUniswapV2Router02SwapOnly(_router);
+        swap = ISwap(_swap);
+        minter = IMinter(_minter);
+
+        stakingToken0 = IERC20(swap.getToken(0));
+        stakingToken1 = IERC20(swap.getToken(1));
+
+        for (uint256 i = 0; i < _rewardsTokens.length; i++) {
+            rewardsTokens.push(IERC20(_rewardsTokens[i]));
+        }
+
+        require(
+            _celoNativeStakingTokenIndex <= 1,
+            "Index of Celo-native token in the LP must be 0 or 1"
+        );
+        celoNativeStakingTokenIndex = _celoNativeStakingTokenIndex;
+        bridgedStakingTokenIndex = 1 - celoNativeStakingTokenIndex;
+
+        if (celoNativeStakingTokenIndex == 0) {
+            celoNativeStakingToken = stakingToken0;
+        } else {
+            celoNativeStakingToken = stakingToken1;
+        }
+    }
+
+    function _deposit(uint256 _lpAmount) internal override whenNotPaused {
+        require(_lpAmount > 0, "Cannot invest in farm because lpAmount is 0");
+        stakingToken.safeIncreaseAllowance(address(liquidityGauge), _lpAmount);
+        liquidityGauge.deposit(_lpAmount);
+    }
+
+    function _withdraw(uint256 _lpAmount) internal override {
+        liquidityGauge.withdraw(_lpAmount);
+    }
+
+    function _claimRewards() internal whenNotPaused {
+        minter.mint(address(liquidityGauge));
+        liquidityGauge.claim_rewards(address(this));
```

Comment:
huh, that's weird

Reply:
not how I would've expected this to work

Reply:
(commenting on how Mobius works here)

Reply:
Yeah, if you farm on Mobius, you'll notice that you have to perform two separate TXs to get all your rewards. Not a great UX imo.